pkg/isoeditor/rhcos.go

```diff
@@ -72,8 +70,40 @@ func (e *rhcosEditor) CreateMinimalISOTemplate(fullISOPath, rootFSURL, arch, min
 	return nil
 }
 
-func embedInitrdPlaceholders(extractDir string) error {
-	f, err := os.Create(filepath.Join(extractDir, ramDiskImagePath))
+// CreateMinimalISOAgentInstaller Creates the minimal iso by removing the rootfs and adding the url for agent installer
+func CreateMinimalISOAgentInstaller(agentIsoFile, extractDir, volumeID, rootFSURL, arch, rdImagePath string) error {
+
+	if err := os.Remove(filepath.Join(extractDir, "images/pxeboot/rootfs.img")); err != nil {
+		return err
+	}
+
+	if err := embedInitrdPlaceholders(extractDir, rdImagePath); err != nil {
+		log.WithError(err).Warnf("Failed to embed initrd placeholders")
+		return err
+	}
+
+	if err := fixGrubConfig(rootFSURL, extractDir, rdImagePath); err != nil {
+		log.WithError(err).Warnf("Failed to edit grub config")
+		return err
+	}
+
+	err := fixIsolinuxConfigForPpc64le(arch, rootFSURL, extractDir)
+	if err != nil {
+		return err
+	}
+
+	if err := Create(agentIsoFile, extractDir, volumeID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func embedInitrdPlaceholders(extractDir, rdImagePath string) error {
+	if rdImagePath == "" {
+		rdImagePath = ramDiskImagePath
```

Comment:
Instead of this pass `ramDiskImagePath` from the callers that need it.
Don't mess around with empty/default parameters.

Reply:
Restructured.  I think the agent installer doesn't care what the value of ramDiskImagePath is so we can use the default one here and not ask agent installer to pass